File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a,b) -> (a[0] == b[0])?(a[1] - b[1]):(a[0] - b[0]));
        List<int[]> _intervals = new ArrayList();
        int[] cur = intervals[0];
        for(int i=1; i<intervals.length; i++){
            if(cur[1] >= intervals[i][0]){
                cur[1] = Math.max(cur[1], intervals[i][1]);
            }else{
                _intervals.add(cur);
                cur = intervals[i];
            }
        }
        _intervals.add(cur);
        
        return _intervals.toArray(new int[_intervals.size()][]);        
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res: List[List[int]] = []
        intervals.sort()
        cur: List[int] = intervals[0]
        for i in range(1, len(intervals)):
            if(intervals[i][0] <= cur[1]):
                cur[1] = max(cur[1], intervals[i][1])
            else:
                res.append(cur)
                cur = intervals[i]
        res.append(cur)
        return res
        